### PR TITLE
Add support for LocalConnectTimeoutMs and LocalRequestTimeoutMs on service-defaults CRD

### PR DESCRIPTION
**Pull request by erdanzhang**
Note original PR https://github.com/hashicorp/consul-k8s/pull/1642, but I couldn't get CLA works since I signed it with a different email than the author's, and seems cannot be changed.

Changes proposed in this PR:

Add maxInboundConnections to service-defaults CRD
fix https://github.com/hashicorp/consul-k8s/issues/1641
Checklist:

Tests added
Reference of similar PR before adding entries to serviceDefault:
https://github.com/hashicorp/consul-k8s/pull/1437

### CODE REVIEW

**Comment by erdanzhang** [comment]
@thisisnotashwin Per your comments in the original PR:
```
The changes to this file should be autogenerated by running make ctrl-generate ctrl-manifests
```

Auto-gen code is cool.
After I ran make ctrl-generate ctrl-manifests, there were few other files that got changed as well.
```
	modified:   charts/consul/templates/crd-peeringacceptors.yaml
	modified:   charts/consul/templates/crd-peeringdialers.yaml
	modified:   charts/consul/templates/crd-servicedefaults.yaml
	modified:   control-plane/api/v1alpha1/zz_generated.deepcopy.go
	modified:   control-plane/config/crd/bases/consul.hashicorp.com_servicedefaults.yaml
```
Could you help double-check if the changes are intended?

File: charts/consul/templates/crd-peeringacceptors.yaml [hunk 1]
```diff
@@ -1,4 +1,4 @@
-{{- if and .Values.connectInject.enabled .Values.global.peering.enabled }}
+{{- if .Values.controller.enabled }}
```
Comment (thisisnotashwin) [inline charts/consul/templates/crd-peeringacceptors.yaml:1]: Can this be reverted from this file and the one below it?

**Review by thisisnotashwin — COMMENTED** [review]
This looks great! Just holding off of the approval as we need to revert the `if` conditions on the acceptors and dialers yaml files.

File: charts/consul/templates/crd-peeringacceptors.yaml (hunk 1 above)
Comment (t-eckert) [reply charts/consul/templates/crd-peeringacceptors.yaml:1]: +1

**Review by t-eckert — APPROVED** [review]
Excellent work! Thank you for doing this!

**Review by t-eckert — APPROVED** [review]

File: charts/consul/templates/crd-peeringacceptors.yaml (hunk 1 above)
Comment (erdanzhang) [reply charts/consul/templates/crd-peeringacceptors.yaml]: @thisisnotashwin Thanks for taking the time to review.
I reverted the change to 
```
charts/consul/templates/crd-peeringacceptors.yaml
charts/consul/templates/crd-peeringdialers.yaml
```

And kept the change to 

```
	modified:   charts/consul/templates/crd-servicedefaults.yaml
	modified:   control-plane/api/v1alpha1/zz_generated.deepcopy.go
	modified:   control-plane/config/crd/bases/consul.hashicorp.com_servicedefaults.yaml
```

For file, control-plane/api/v1alpha1/zz_generated.deepcopy.go, not sure if the change looks good
image

Could you help confirm if the changes are correct?

Comment (thisisnotashwin) [reply charts/consul/templates/crd-peeringacceptors.yaml]: Hey @erdanzhang !! Those changes look right!! I really appreciate you reaching out to confirm it!! Also, thank you so much, again, for your contribution!!

**Review by thisisnotashwin — APPROVED** [review]

**Pull request merged**